compiler/src/dotty/tools/dotc/staging/CrossStageSafety.scala

```diff
@@ -104,6 +105,19 @@ class CrossStageSafety extends TreeMapWithStages {
       case _: DefDef if tree.symbol.isInlineMethod =>
         tree
 
+      case tree: CaseDef if level == 0 =>
```

Comment:
The important this in this transformation is to introduce the new pattern variables (implicit `Type[t]`) in the pattern scope. These are needed to transform the RHS of the `case`. We need to encode the patterns before we look for the patterns variables to make sure they are in the context when needed.

An alternative would be implement this with a variant of the `CaseDef` case in `TreeMapWithImplicits` that collects the scope a second time after transforming the pattern. This would also need to include the context update in the `CaseDef` case of `TreeMapWithStages`. I chose not to do this because any change in the super-classes would need to be reflected here. This might be missed easily.